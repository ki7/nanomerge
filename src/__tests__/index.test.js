var Merge = require("../merge");

jest.mock("../merge", function() {
  var FakeMerge = jest.fn();

  FakeMerge.prototype.merge = jest.fn();

  return FakeMerge;
});

var nanomerge = require("../.");

afterEach(function() {
  Merge.mockClear();
  Merge.prototype.merge.mockClear();
});

describe("#()", function() {
  it("Must to create Merge", function() {
    expect(Merge.mock.calls).toMatchSnapshot();
  });

  it("Must run Merge.merge", function() {
    nanomerge({}, {}, {});
    nanomerge([], []);
    nanomerge([], [], {});

    expect(Merge.prototype.merge.mock.calls).toMatchSnapshot(); // Must run .merge
  });
});

describe("#create()", function() {
  it("Must be function", function() {
    expect(typeof nanomerge.create).toBe("function");
  });

  it("Must be throw error if config not equal object", function() {
    expect(nanomerge.create).toThrow();
    expect(nanomerge.create.bind(null, 4)).toThrow();
    expect(nanomerge.create.bind(null, null)).toThrow();
  });

  it("Must not create new Merge immediately", function() {
    expect(Merge.mock.calls).toMatchSnapshot();
  });

  it("Must create new Merge by a call", function() {
    nanomerge.create({});

    expect(Merge.mock.calls).toMatchSnapshot();
  });

  it("Must run Merge.merge", function() {
    var merger = nanomerge.create({});

    merger({}, {}, {});
    merger([], []);
    merger([], [], {});

    expect(Merge.prototype.merge.mock.calls).toMatchSnapshot(); // Must run .merge
  });
});
