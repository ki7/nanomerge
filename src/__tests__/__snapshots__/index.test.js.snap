// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#() Must run Merge.merge 1`] = `
Array [
  Array [
    Object {},
    Object {},
    Object {},
  ],
  Array [
    Array [],
    Array [],
  ],
  Array [
    Array [],
    Array [],
    Object {},
  ],
]
`;

exports[`#() Must to create Merge 1`] = `
Array [
  Array [
    Object {},
  ],
]
`;

exports[`#create() Must create new Merge by a call 1`] = `
Array [
  Array [
    Object {},
  ],
]
`;

exports[`#create() Must not create new Merge immediately 1`] = `Array []`;

exports[`#create() Must run Merge.merge 1`] = `
Array [
  Array [
    Object {},
    Object {},
    Object {},
  ],
  Array [
    Array [],
    Array [],
  ],
  Array [
    Array [],
    Array [],
    Object {},
  ],
]
`;
